open_fdd/gateway/local_codex_cli.py

```
@@ -0,0 +1,413 @@
+        cmd.extend(["--model", str(model).strip()])
+    cmd.extend(["--skip-git-repo-check", "--color", "never", "-"])
+    return cmd
+
+
+def run_npm_install_codex_global(*, timeout_s: int | None = None) -> dict[str, Any]:
+    """Run ``npm install -g @openai/codex`` on the bridge host (operator-local desktop bridge)."""
+    t = timeout_s if timeout_s is not None else safe_int_from_env("OFDD_NPM_INSTALL_CODEX_TIMEOUT_S", 600)
+    t = max(60, min(t, 3600))
+    try:
+        cp = subprocess.run(
+            ["npm", "install", "-g", "@openai/codex"],
```

Ruff (0.15.12)

[error] Starting a process with a partial executable path

(S607)